js/views/file_input_view.js

```diff
@@ -301,6 +301,16 @@
         return;
       }
 
+      // Save temporary copy of file to disk.
+      const arrayBuffer = await new Promise((resolve) => {
+        const fileReader = new FileReader();
+        fileReader.onload = () => resolve(fileReader.result);
+        fileReader.readAsArrayBuffer(originalFile);
+      });
+      const ext = originalFile.type.split('/')[1];
```

Comment:
What is this line trying to do, get the file extension? If so, it should split on '.' and protect against files with no extension. Either way, I suspect that we shouldn't put file extensions on our temporary files at all.